examples/ipfs-kad/src/main.rs

```diff
@@ -110,3 +126,62 @@ async fn main() -> Result<(), Box<dyn Error>> {
 
     Ok(())
 }
+
+async fn insert_pk_record<TStore: kad::store::RecordStore + Send + 'static>(
+    swarm: &mut Swarm<kad::Behaviour<TStore>>,
+    local_peer_id: PeerId,
+    local_key: Keypair,
+) -> Result<(), Box<dyn Error>> {
+    let mut pk_record_key = vec![];
+    pk_record_key.put_slice("/pk/".as_bytes());
+    pk_record_key.put_slice(local_peer_id.to_bytes().as_slice());
+
+    println!("Putting PK record into the DHT");
+
+    let mut pk_record = kad::Record::new(&pk_record_key, local_key.public().encode_protobuf());
+
+    pk_record.publisher = Some(local_peer_id);
+    pk_record.expires = Some(Instant::now().add(Duration::from_secs(60)));
+
+    swarm
+        .behaviour_mut()
+        .put_record(pk_record, kad::Quorum::N(NonZeroUsize::new(3).unwrap()))?;
+
+    loop {
+        let event = swarm.select_next_some().await;
+        if let SwarmEvent::Behaviour(kad::Event::OutboundQueryProgressed {
+            result: kad::QueryResult::PutRecord(result),
+            ..
+        }) = event
+        {
+            match result {
+                Ok(_) => {
+                    println!("Successfully inserted the PK record");
+                }
+                Err(err) => {
+                    println!("Failed to insert the PK record {err}");
+                }
+            }
+
+            break;
+        }
+    }
+
+    Ok(())
+}
+
+#[derive(Parser, Debug)]
+#[clap(name = "libp2p Kademlia DHT example")]
+struct Opt {
+    #[clap(subcommand)]
+    argument: CliArgument,
+}
+
+#[derive(Debug, Parser)]
+enum CliArgument {
+    GetPeers {
+        #[clap(long)]
+        peer_id: Option<String>,
```

Comment:
```suggestion
        peer_id: Option<PeerId>,
```

Let `clap` do the heavy lifting of parsing :)